stack/ArrayStack.java

```diff
@@ -35,52 +33,125 @@ public ArrayStack(int size){
         this.stack = (T[]) new Object[size];
     }
 
+    /** 
+     * Test if stack empty
+     * @return true if empty, false otherwise
+     */
     public boolean isEmpty(){
+        if(this.top == -1){
+            return true;
+        } 
         return false;
     }
 
+    /**
+     * Return item at top of stack without removing it
+     * @return item at the top of stack
+     * @throws exception if stack is empty
+     */
     public T peek() throws Exception {
-    
-        return null;
+        if(this.isEmpty()){
+            throw new Exception("Stack is empty");
+        }
+        return this.stack[this.top];
     }
 
+    /**
+     * Remove item at top of stack
+     * @return the item at the top of the stack
+     * @throws exception if stack is empty
+     */
     public T pop() throws Exception {
-        return null;
+        if(this.isEmpty()){
+            throw new Exception("Stack is empty");
+        }
+
+        return this.stack[this.top--];
     }
 
+    /** 
+     * Add an item to the top of the stack
+     * @return the item
+     */
     public T push(T item){
-        return null;
+        if(this.isFull()){
+            
+            this.extendCapacity();
+        }
+        
+        this.stack[++this.top] = item;
+        
+        return item;
     } 
+
+    /**
+     * Search for item in the stack
+     * @return the distance from the top of the stack of the closest to the top occurence
+     *          -1 if item does not occur on stack
+     * @throws exception if stack is empty 
+     */
     public int search(T item) throws Exception {

```

Comment:
What does this function do? I haven't seen `search` as a stack function before.

Reply:
I followed - https://docs.oracle.com/javase/7/docs/api/java/util/Stack.html